src/solvers/flattening/arrays.cpp

```diff
@@ -334,10 +325,8 @@ void arrayst::add_array_Ackermann_constraints()
 
           if(indices_equal_lit!=const_literal(false))
           {
-            index_exprt index_expr1;
-            index_expr1.type()=ns.follow(arrays[i].type()).subtype();
-            index_expr1.array()=arrays[i];
-            index_expr1.index()=*i1;
+            const typet &subtype=ns.follow(arrays[i].type()).subtype();
```

Comment:
No, the arrays handled by the array theory solver won't get touched by bitvector conversion.

Yes, this code should be obsolete; symbol types are only to be generated for structs (and should eventually get replaced by struct_tag types).